app/uk/gov/hmrc/gform/formtemplate/FormTemplatesControllerRequestHandler.scala

```diff
@@ -164,6 +164,48 @@ object FormTemplatesControllerRequestHandler {
         Json.obj()
       )
 
+    def transformProgressIndicatiorInArray =
+      __.read[JsArray].map { pages =>
+        JsArray(
+          pages.value.map {
+            case page: JsObject if page.value.contains("progressIndicator") =>
```

Comment:
Why is this condition needed. Either you can do the transformation or you can't. 

Writing `transformProgressIndicator` in transformer style would allow to `transformProgressIndicatiorInArray` more concisely in terms `Reads` api, rather than this mix of style right now.